l10n_br_sale/models/sale.py

```diff
@@ -0,0 +1,376 @@
+            kwargs = {
+                'partner_id': self.partner_id.id,
+                'partner_invoice_id': self.partner_invoice_id.id,
+                'fiscal_category_id': self.fiscal_category_id.id,
+                'company_id': self.company_id.id,
+                'context': self.env.context
+            }
+            result = self.env[
+                'account.fiscal.position.rule'].apply_fiscal_mapping(
+                    result, **kwargs)
+            self.fiscal_position = result['value'].get('fiscal_position')
+
+    @api.model
+    def _fiscal_comment(self, order):
+        fp_comment = []
+        fp_ids = []
+
+        for line in order.order_line:
+            if line.fiscal_position and \
+                    line.fiscal_position.inv_copy_note and \
+                    line.fiscal_position.note:
+                if not line.fiscal_position.id in fp_ids:
+                    fp_comment.append(line.fiscal_position.note)
+                    fp_ids.append(line.fiscal_position.id)
+
+        return fp_comment
+
+    @api.model
+    def _prepare_invoice(self, order, lines):
+        """Prepare the dict of values to create the new invoice for a
+           sale order. This method may be overridden to implement custom
+           invoice generation (making sure to call super() to establish
+           a clean extension chain).
+
+           :param browse_record order: sale.order record to invoice
+           :param list(int) line: list of invoice line IDs that must be
+                                  attached to the invoice
+           :return: dict of value to create() the invoice
+        """
+        result = super(SaleOrder, self)._prepare_invoice(order, lines)
+        context = self.env.context
+
+        inv_lines = self.env['account.invoice.line'].browse(lines)
+
+        if (context.get('fiscal_type') == 'service' and
+                inv_lines and inv_lines[0].fiscal_category_id):
+            fiscal_category_id = inv_lines[0].fiscal_category_id.id
+            result['fiscal_position'] = inv_lines[0].fiscal_position.id
+        else:
+            fiscal_category_id = order.fiscal_category_id.id
+
+        if fiscal_category_id:
+            fiscal_category = self.env[
+                'l10n_br_account.fiscal.category'].browse(fiscal_category_id)
+            if fiscal_category:
+                result['journal_id'] = fiscal_category[0].property_journal.id
+
+        result['partner_shipping_id'] = order.partner_shipping_id.id
+
+        comment = []
+        if order.note and order.copy_note:
+            comment.append(order.note)
+
+        fiscal_comment = self._fiscal_comment(order)
+        result['comment'] = " - ".join(comment)
+        result['fiscal_comment'] = " - ".join(fiscal_comment)
+        result['fiscal_category_id'] = fiscal_category_id
+
+        return result
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    @api.one
```

Comment:
forget that, see https://github.com/OCA/l10n-brazil/pull/145#discussion_r45286291